python/cudf/cudf/core/dtypes.py

```diff
@@ -221,14 +221,51 @@ class DecimalDtype(ExtensionDtype):
 
     name = "decimal"
     _metadata = ("precision", "scale")
+    _MAX_PRECISION = np.floor(np.log10(np.iinfo("int64").max))
 
     def __init__(self, precision, scale):
+        """
+        Parameters
+        ----------
+        precision : int
+            The total number of digits in each value of this dtype
+        scale : int
+            The scale of the DecimalDtype. See Notes below.
+
+        Notes
+        -----
+            When the scale is positive:
+              - numbers with fractional parts (e.g., 0.0042) can be represented
+              - the scale is the total number of digits to the right of the
+                decimal point
+            When the scale is negative:
+              - only multiples of powers of 10 (including 10**0) can be
+                represented (e.g., 1729, 4200, 1000000)
+              - the scale represents the number of trailing zeros in the value.
+            For example, 42 is representable with precision=2 and scale=0.
+            13.0051 is representable with precision=6 and scale=4,
+            and *not* representable with precision<6 or scale<4.
+        """
+        if precision > self._MAX_PRECISION:
+            raise ValueError(
+                f"Cannot construct a {type(self).__name__}"
+                f" with precision > {self._MAX_PRECISION}"
+            )
         self._typ = pa.decimal128(precision, scale)
 
     @property
     def precision(self):
         return self._typ.precision
 
+    @precision.setter
+    def precision(self, value):
+        if value > self._MAX_PRECISION:
+            raise ValueError(
+                f"Cannot construct a {type(self).__name__}"
+                f" with precision > {self._MAX_PRECISION}"
+            )
+        self._typ = pa.decimal128(precision=value, scale=self.scale)
+
     @property
     def scale(self):
         return self._typ.scale
```

Comment:
Should we validate this value similar to what we'll be doing with `precision`?

Reply:
What are the constraints on the `scale`? Is `scale <= precision` sufficient?

Reply:
I believe you need to do `abs(scale) <= precision` since `scale` can be negative as well. As far as I can tell it's not legal to have an absolute larger scale than the precision but I could be mistaken.

Reply:
I don't think it makes sense to have `abs(scale) <= precision`, but FWIW, arrow allows it.

Reply:
Based on discussion with @codereport, we now check that `abs(scale) <= precision`. The `precision` is interpreted to be the _total_ number of digits (not the number of _significant_ digits).